tests/simpleKVBC/TesterReplica/setup.cpp

```diff
@@ -151,6 +164,20 @@ std::unique_ptr<TestSetup> TestSetup::ParseArgs(int argc, char** argv) {
 
     if (keysFilePrefix.empty()) throw std::runtime_error("missing --key-file-prefix");
 
+    // If -p and -t are set, enable clientTransactionSigningEnabled. If only one of them is set, throw an error
+    if ( !principalsMapping.empty() ) {
+      if ( !txnSigningKeysPath.empty() ) {
+        replicaConfig.clientTransactionSigningEnabled = true;
+      }
+      else {
+        throw std::runtime_error("Expected param txn-signing-key-path to be set when principals-mapping is set");
+      }
+    }
+
+    if (replicaConfig.clientBatchingEnabled) {
```

Comment:
I assumed you wanted to check replicaConfig.clientTransactionSigningEnabled?
Why not adding this 1 liner after line 170? (after fixes you make).